### ISSUE TITLE
Support Python 3.9

### ISSUE THREAD

**Issue by csadorf** [aiidateam/aiida-core#4302]
According to the [Python 3.9 RC release notes](https://www.python.org/downloads/release/python-390rc1/):

> We strongly encourage maintainers of third-party Python projects to prepare their projects for 3.9 compatibility during this phase. As always, report any issues to the Python bug tracker.

I will track preparations to support Python 3.9 on this issue.

**Comment by csadorf** [comment]
Commands for testing
- [ ] `docker run python:3.9.0rc1 pip install aiida-core` 
- [ ] `docker run python:3.9.0rc1 pip install aiida-core[all]` 
 - scipy fails to build because of lapack missing

**Comment by csadorf** [comment]
Keeping track of things here: https://github.com/aiidateam/aiida-core/pull/4301

**Comment by greschd** [comment]
@csadorf what's the status on this? AFAIK 3.9 should be released on 2020-10-05, three days from now..

**Comment by csadorf** [comment]
> @csadorf what's the status on this? AFAIK 3.9 should be released on 2020-10-05, three days from now..

No update yet. I assume that we will have to wait for dependencies to support Python 3.9 before we can move forward with this. Obviously we should aim for support as soon as possible, but there is nothing we can do if packages like pandas can't be installed yet.

**Comment by csadorf** [comment]
This issue was resolved with the merge of #4301 .

**Comment by giovannipizzi** [comment]
I found at least two more issues with py 3.9 compatibility.

First, I created this: DropD/reentry#57 that needs to be fixed first.

Second: I have (from `pip freeze`): `SQLAlchemy-Utils==0.34.2` that matches our requirements in the setup.json (`sqlalchemy-utils~=0.34.2`). 
However, with this requirement I get:
```
tests/transports/test_all_plugins.py::TestBasicFunctionality::test_is_open
  /Users/pizzi/.virtualenvs/aiida-dev/lib/python3.8/site-packages/sqlalchemy_utils/observer.py:176: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    from collections import defaultdict, Iterable, namedtuple
```
I think this has been fixed in [this PR of them](https://github.com/kvesteri/sqlalchemy-utils/pull/397) and, looking [at the corresponding commit](https://github.com/kvesteri/sqlalchemy-utils/commit/45f4e4a3d83f7e45cdf449c7c2c4bfab659d47a7), this should have been released from 0.35.0 (see list of tags in the blue bar at the top of the page from GitHub). So we should update the dependency?

Probably this second issues is not caught in new venvs since the most recent version is installed, but I had an older venv.

@csadorf I'm reopening this issue, but of course feel free to open another one instead moving my comments there.
